text/0104-string-parameterization-output-spec.md

```diff
@@ -0,0 +1,293 @@
+
+```
+GET http://someservice.com/users/{userName}
+```
+
+**Note:** The parameterizer should replace the smallest possible substring. e.g., the URL `/service/id-17` should be parameterized as `/service/id-{userId}` rather than `/service/{userId}`.
+
+**Note:** The parameterizer must output parameter names in `camelCase` format.
+
+### Unknown Route Segments
+
+If a parameterizer does not have knowledge of the route structure, but it determines that a value within a URL is a dynamic segment, **it must replace the value with a single `*` character.**
+
+Given:
+
+```
+GET http://someservice.com/users/tom/info
+```
+
+If the parameterizer determines that `tom` is a parameter of unknown significance, it must return:
+
+```
+GET http://someservice.com/users/*/info
+```
+
+**Note:** The parameterizer must preserve the original URL structure as much as possible. e.g., the URL `https://service.io/data/sentry/frontend/info` where both `sentry` and `frontend` are dynamic route segments, the parameterizer must return `https://service.io/data/*/*/info` rather than `https://service.io/data/*/info`.
+
+### Subdomains
+
+Subdomain can contain parameters, too. For example, in the URL `https://company.someservice.io/` the string `company` might correspond to a client name, and is a dynamic string.
+
+If a parameterizer determines that a section of a domain string is dynamic, **it must replace the segment with a single `*` character.**
+
+Given:
+
+```
+GET http://host1.someservice.com/users/
+```
+
+If the parameterizer determines that `host1` is a parameter, it must return:
+
+```
+GET http://*.someservice.com/users/
+```
+
+**Note:** The parameterizer must not alter domains or hostnames in any other way.
+
+### Port Numbers
+
+**If the URL contains a port number, the parameterizer should not parameterize it.**
+
+Given:
+
+```
+GET http://someservice.com:8080/
+```
+
+Return:
+
+```
+GET http://someservice.com:8080/
+```
+
+### Auth Strings
+
+**If the URL contains authentication strings, the parameterizer must replace them with a `*` character.**
+
+Given:
+
+```
+GET http://admin:password@someservice.com/
+```
+
+Return:
+
+```
+GET http://*:*@someservice.com/
+```
+
+## Parameterizing Key-Value Store Key Names
+
+Key-value stores like Redis store arbitrary data, and are accessed by a “key”, which is a string. It is a common convention to namespace these keys, delimit them with a `:` or `.` character, and put parameters between the namespaces. e.g., the order IDs of a given user might be stored under the key `organizations:18:users:9982:orders`. `18` and `9982` are parameters. `organizations`, `users`, and `orders` are namespaces.
+
+## String Keys
+
+**For parameters in key names, the parameterizer must replace each parameter with a single `*` character.**
+
+Given:
+
+```
+SADD orgs:17:emails jane@myorg.com
+```
+
+Return:
+
+```
+SADD orgs:*:emails jane@myorg.com
+```
+
+## Structured JSON
```

Comment:
How are structured JSONs sent? Is this part of `span.data`?

Reply:
The intention here is to cover cases where JSON data is stringified into a string field. For example, Snuba spans do this with their descriptions. This is not a common case as far as I know!

Reply:
Technically, the example below isn't valid JSON. Do we have a spec of what the language we're looking for looks like?